client/src/connection/rest/RestLibraryAdapter.ts

```diff
@@ -0,0 +1,252 @@
+// Copyright © 2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { ProgressLocation, l10n, window } from "vscode";
+
+import { AxiosResponse } from "axios";
+
+import { getSession } from "..";
+import {
+  LibraryAdapter,
+  LibraryItem,
+  TableData,
+} from "../../components/LibraryNavigator/types";
+import { LogFn as LogChannelFn } from "../../components/LogChannel";
+import {
+  ColumnCollection,
+  DataAccessApi,
+  RowCollection,
+  TableInfo,
+} from "./api/compute";
+import { getApiConfig } from "./common";
+
+const requestOptions = {
+  headers: { Accept: "application/vnd.sas.collection+json" },
+};
+
+class RestLibraryAdapter implements LibraryAdapter {
+  protected dataAccessApi: ReturnType<typeof DataAccessApi>;
+  protected sessionId: string;
+
+  public async connect(): Promise<void> {
+    const session = getSession();
+    session.onLogFn = LogChannelFn;
+
+    await window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: l10n.t("Connecting to SAS session..."),
+      },
+      async () => {
+        await session.setup();
+      },
+    );
```

Comment:
I'm glad you mentioned this one. I _do_ need to think of a better way to address this. Ideally, I think it might not be a bad idea to do a better job isolating vscode-isms from our code in "deeper" layers

Reply:
Done in https://github.com/sassoftware/vscode-sas-extension/pull/733/commits/586f7636bb230e75b7bd39b4458020f006df3c88